src/game/AI/ScriptDevAI/scripts/northrend/ulduar/ulduar/boss_ignis.cpp

```diff
@@ -82,300 +85,306 @@ enum
         {
-            if (uiDamage > 5000)
-            {
-                DoCastSpellIfCan(m_creature, SPELL_SHATTER, CAST_TRIGGERED);
-                DoCastSpellIfCan(m_creature, SPELL_STRENGTH_REMOVE, CAST_TRIGGERED);
+            m_creature->RemoveAllAurasOnDeath();
+            DoCastSpellIfCan(nullptr, SPELL_SHATTER, CAST_TRIGGERED);
+            DoCastSpellIfCan(nullptr, SPELL_STRENGTH_REMOVE, CAST_TRIGGERED);
 
-                m_creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNINTERACTIBLE);
-                m_creature->GetMotionMaster()->MoveIdle();
-                m_bHasShattered = true;
-            }
+            m_creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNINTERACTIBLE);
+            m_creature->SetImmuneToPlayer(true);
+            m_creature->SetImmuneToNPC(true);
+            m_creature->GetMotionMaster()->MoveIdle();
+            SetMeleeEnabled(false);
+            SetCombatMovement(false);
+            m_hasShattered = true;
         }
     }
+};
 
-    void SpellHit(Unit* pCaster, const SpellEntry* pSpell) override
+bool EffectScriptEffectCreature_npc_iron_construct(Unit* caster, uint32 spellId, SpellEffectIndex EffIdx, Creature* creatureTarget, ObjectGuid /*originalCasterGuid*/)
+{
+    if (spellId == SPELL_WATER_EFFECT && EffIdx == EFFECT_INDEX_0 && creatureTarget->GetEntry() == NPC_IRON_CONSTRUCT)
     {
-        if (pSpell->Id == SPELL_HEAT)
-        {
-            if (SpellAuraHolder* pHeatAura = m_creature->GetSpellAuraHolder(SPELL_HEAT))
-            {
-                if (pHeatAura && pHeatAura->GetStackAmount() == MAX_HEAT_STACKS)
-                    DoCastSpellIfCan(m_creature, SPELL_MOLTEN);
-            }
-        }
+        // chill the iron construct if molten (effect handled in core)
+        if (creatureTarget->HasAura(SPELL_MOLTEN))
+            creatureTarget->CastSpell(nullptr, SPELL_CHILL, TRIGGERED_OLD_TRIGGERED);
+
+        return true;
     }
 
-    void UpdateAI(const uint32 /*uiDiff*/) override
-    {
-        if (!m_creature->SelectHostileTarget() || !m_creature->GetVictim())
-            return;
+    return false;
+}
 
-        // stop attacking after shattered
-        if (m_bHasShattered)
-            return;
+/*######
+## npc_scorch
+######*/
+
+// TODO Remove this 'script' when combat can be proper prevented from core-side
+struct npc_scorchAI : public ScriptedAI
+{
+    npc_scorchAI(Creature* creature) : ScriptedAI(creature) { Reset(); }
 
-        DoMeleeAttackIfReady();
+    void Reset() override
+    {
+        SetMeleeEnabled(false);
+        SetCombatMovement(false);
+        SetReactState(REACT_PASSIVE);
+        m_creature->ForcedDespawn(50 * IN_MILLISECONDS);
     }
 };
 
-UnitAI* GetAI_npc_iron_construct(Creature* pCreature)
+bool EffectScriptEffectCreature_npc_scorch(Unit* caster, uint32 spellId, SpellEffectIndex EffIdx, Creature* creatureTarget, ObjectGuid /*originalCasterGuid*/)
```

Comment:
Rewrite this using a new style spell script instead of this. Long term goal of removing this from core. Its same as OnEffectExecute